src/main/java/org/opentripplanner/graph_builder/module/ned/ElevationModule.java

```diff
@@ -115,28 +142,21 @@ public ElevationModule(
 
     @Override
     public void buildGraph(Graph graph, GraphBuilderModuleSummary graphBuilderModuleSummary) {
+        this.graph = graph;
         GraphBuilderTaskSummary demPrepareTask = graphBuilderModuleSummary.addSubTask(
             "fetch and prepare elevation data"
         );
         log.info(demPrepareTask.start());
 
         gridCoverageFactory.setGraph(graph);
-        Coverage gridCov = gridCoverageFactory.getGridCoverage();
-
-        // If gridCov is a GridCoverage2D, apply a bilinear interpolator. Otherwise, just use the
-        // coverage as is (note: UnifiedGridCoverages created by NEDGridCoverageFactoryImpl handle
-        // interpolation internally)
-        coverage = (gridCov instanceof GridCoverage2D) ? Interpolator2D.create(
-                (GridCoverage2D) gridCov, new InterpolationBilinear()) : gridCov;
-        try {
-            transformer = CRS.findMathTransform(
-                GeometryUtils.WGS84_XY,
-                coverage.getCoordinateReferenceSystem(),
-                true
-            );
-        } catch (FactoryException e) {
-            log.error("Could not find an appropriate transformer!");
-            throw new RuntimeException(e);
+        if (transformCoordinates) {
+            try {
+                transformer = CRS
+                    .findMathTransform(GeometryUtils.WGS84_XY, getCoverage().getCoordinateReferenceSystem(), true);
+            } catch (FactoryException e) {
+                log.error("Could not find an appropriate transformer!");
+                throw new RuntimeException(e);
+            }
```

Comment:
I'm fairly certain there is no need to do any transformation of any kind. [`Interpolator2D#evaluate`](https://github.com/geotools/geotools/blob/59f0026c5bc1848fa35d102235571bfcb38794c7/modules/library/coverage/src/main/java/org/geotools/coverage/grid/Interpolator2D.java#L338) already handles this as far as I can tell.

Reply:
This is required during testing, otherwise, a downstream library will raise an assertion error.